subsys/dfu/Kconfig

```diff
@@ -14,20 +14,34 @@ menuconfig IMG_MANAGER
 
 if IMG_MANAGER
 
-choice
-	prompt "Image manager"
-	default MCUBOOT_IMG_MANAGER
+config FLASH_IMG
+	bool "Flash Image API"
+	select FLASH_MAP
+
+config IMG_BLOCK_BUF_SIZE
+	int "Image writer buffer size"
+	depends on FLASH_IMG
+	default 512
 	help
-	  Choice support for managing DFU image.
-	  So far only mcuboot support is available.
+	  Size (in Bytes) of buffer for image writer. Must be a multiple of
+	  the access alignment required by used flash driver.
+
+config IMG_ERASE_PROGRESSIVELY
+	bool "Erase flash progressively when receiving new firmware"
+	depends on FLASH_IMG
+	select FLASH_PAGE_LAYOUT
+	help
+	  If enabled, flash is erased as necessary when receiving new firmware,
+	  instead of erasing the whole image slot at once. This is necessary
+	  on some hardware that has long erase times, to prevent long wait
+	  times at the beginning of the DFU process.
 
 config MCUBOOT_IMG_MANAGER
 	bool "Image manager for mcuboot"
-	select FLASH_MAP
+	select FLASH_IMG
```

Comment:
Is this for backward compatibility? -I believe this not really required for meet dependency between module. Maybe better to make `FLASH_IMG=y` by default depend on this property?